iroha_p2p/src/tests/mod.rs

```diff
@@ -0,0 +1,65 @@
+#![allow(clippy::unwrap_used)]
+use iroha_actor::{broker::*, prelude::*};
+use iroha_logger::{config::LoggerConfiguration, info};
+use parity_scale_codec::{Decode, Encode};
+use tokio::time::Duration;
+use ursa::{encryption::symm::prelude::ChaCha20Poly1305, kex::x25519::X25519Sha256};
+
+use super::*;
+use crate::peer::PeerId;
+
+#[derive(iroha_actor::Message, Clone, Debug, Decode, Encode)]
+struct TestMessage(String);
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+async fn network_create() {
```

Comment:
I would also add a test, where you actually communicate between several IPs. In your case I think you would need to spawn several tasks and in each of them to start the network with a different IP, then connect all of them together. And check for example if peer at :4433 receives message from ::4432 and so on.

This would test the main use case that we would need.